test/test_tableauauth_model.py

```diff
@@ -0,0 +1,22 @@
+import unittest
+import warnings
+import tableauserverclient as TSC
+
+
+class TableauAuthModelTests(unittest.TestCase):
+    def test_username_password_required(self):
```

Comment:
Don't remove it, just don't name it.

```py
with self.assertRaises():
    TSC.TableauAuth()
```

The `as context_manager` wasn't necessary because you didn't use the object.
The context manager itself is the way to go